ui: fix jobs page crash [BACKPORT 21.1]

vladlos opened the pull request:
while flipping pages on jobs table user could encount page crash. in addition
to that there was a visible delay beetwen data changes that make no sence
since pagination itself is pure ui.
there were a compbination of issues that cause such behavior, this pr contains:
 - on my local setup I got jobs with empty `statement` and `description` for such
case I added fallback to `type` value, to at least leave some value in that column as
it works also as a link and so there should be some clickable value
 - fix for unneeded component rerenders
 - after removing extra rerenders another issue with table component become
visible, wrong use of reselect library, fixed
 - combination last 2 changes fixed issue with slow data refresh
 - in addition fixed font color on `description` value tooltips, it was wrong and barely
visible

Resolves: cockroachdb#62905

Release note (ui): fix jobs page crash while using pagination, improve performance

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65723)
<!-- Reviewable:end -->

vladlos committed:
ui: fix jobs page crash

while flipping pages on jobs table user could encount page crash. in addition
to that there was a visible delay beetwen data changes that make no sence
since pagination itself is pure ui.
there were a compbination of issues that cause such behavior, this pr contains:
- on my local setup I got jobs with empty `statement` and `description` for such
case I added fallback to `type` value, to at least leave some value in that column as
it works also as a link and so there should be some clickable value
- fix for unneeded component rerenders
- after removing extra rerenders another issue with table component become
visible, wrong use of reselect library, fixed
- combination last 2 changes fixed issue with slow data refresh
- in addition fixed font color on `description` value tooltips, it was wrong and barely
visible

Resolves: cockroachdb#62905

Release note (ui): fix jobs page crash while using pagination, improve performance

elkmaster commented:
bors r+
